packages/component_code_gen/templates/generate_apps.py

```diff
@@ -0,0 +1,466 @@
+
+The `methods` property contains auxiliary methods. A `async _makeRequest` method is always required. It contains the code that makes the API request. It takes two arguments: `$ = this` and `opts`. `$` is the context passed by the Pipedream runtime. It should default to `this`. `opts` is an object that contains the parameters of the API request. The `opts` object may contain the following fields: `method`, `path`, `data`, `params`, and `headers`. The `method` field is the HTTP method of the request. The `path` field is the path of the request. The `data` field is the body of the request. The `params` field is the query parameters of the request. The `headers` field is the headers of the request. The `opts` object also contains any other fields that are passed to the `_makeRequest` method. The `_makeRequest` method returns a Promise that resolves to the HTTP response data. There is NO `data` property in the response that contains the data. The data from the HTTP response is returned directly in the response, not in the `data` property.
+
+The axios request uses the authentication method defined by the app. Different apps pass credentials in different places in the HTTP request, e.g. headers, url params, etc. The axios request should use the format from the app docs.
+
+Auxiliary methods, usually for CRUD operations call `_makeRequest` with the appropriate parameters. Please add a few methods for common operations, e.g. get and list. You can also add other methods that you think are useful.
+
+For listing operations, verify how the pagination is done in the API. Also add a method for pagination. This method should be named `paginate`, and the arguments are `fn`, the listing method that will be called, and `...opts`, the parameters of the HTTP request. The method starts with an empty array, and calls the listing method with the parameters. It then checks the response and verifies if there is more data. If it does, it calls itself with the listing method and the parameters for fetching the next set of data. If it doesn't, it returns the array of results.
+
+## Pipedream Platform Axios
+
+Always use the `axios` constructor from the `@pipedream/platform` package, and include the following import at the top of your Node.js code, above the component, in this exact format:
+
+import { axios } from "@pipedream/platform";
+
+You MUST use that import format when importing axios. Do NOT attempt to import any other package like `import axios from "@pipedream/platform/axios"`.
+
+The `axios` constructor takes two arguments:
+
+1. `this` - the context passed by the run method of the component.
```

Comment:
Since `this` or the `$` needs to be passed from the `run` method of the component, shouldn't this section (or the methods section above) articulate how the methods will need to accept the `$` as an argument from the caller?